src/main/java/lukemccon/airdrop/packages/PackagesGui.java

```diff
@@ -0,0 +1,116 @@
+package lukemccon.airdrop.packages;
+
+import lukemccon.airdrop.Airdrop;
+
+import lukemccon.airdrop.exceptions.PackageNotFoundException;
+import org.bukkit.Bukkit;
+import org.bukkit.Material;
+import org.bukkit.entity.HumanEntity;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.inventory.InventoryClickEvent;
+import org.bukkit.event.inventory.InventoryDragEvent;
+import org.bukkit.inventory.Inventory;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * GUI that shows available packages within airdrop
+ */
+public class PackagesGui implements Listener {
+    private final Inventory inv;
+    private final int ROW_SIZE = 9;
+    private final int SAVE_CANCEL_PADDING = 3;
+
+    public int inventorySize;
+
+    public PackagesGui() {
+
+        int inventorySize;
+        int packageCount = PackageManager.getNumberofPackages();
+
+        inventorySize = (int) (ROW_SIZE * Math.ceil(((packageCount + SAVE_CANCEL_PADDING)/ROW_SIZE) + 1 ));
+
+        if (inventorySize < 9) {
+            inventorySize = 9;
+        }
+
+        // Create a new inventory, with no owner (as this isn't a real inventory), a size of nine, called example
+        inv = Bukkit.createInventory(null, inventorySize, "Packages");
+
+        // Put the items into the inventory
+        initializeItems();
+    }
+
+    // You can call this whenever you want to put the items in
+    public void initializeItems() {
+        Set<String> packages = PackageManager.getPackages();
+
+        List<ItemStack> pkglist = packages.stream().map(this::packageGuiItem).toList();
+        pkglist.forEach(inv::addItem);
+    }
+
+    private ItemStack packageGuiItem(String packageName) {
+        Package pkg;
+        double price;
+
+        try {
+           pkg = PackageManager.get(packageName);
+           price = pkg.getPrice();
+        } catch (PackageNotFoundException e) {
+           price = 0.0;
+        }
+        return createGuiItem(Material.CHEST, packageName, "$" + price);
+    }
+
+    // Creates a gui item with a custom name, and description
+    protected ItemStack createGuiItem(final Material material, final String name, final String... lore) {
```

Comment:
It probably doesn't matter too much, but this is the third place this same createGuiItem method has been implemented, all the same way. We should consider extracting this and any other duplicates I didn't notice into like, a helper.

Reply:
Yeah definitely, was just trying to get this all working, I think I have significant refactoring to do in the next release